README.md

```diff
@@ -1,30 +1,80 @@
-# Brzeczyk
-A compiler implemented by TCS UJ students at "Compilers" course in the 2022/23 winter semester.
+# Brzęczyk
 
-## Building
-Project can be built using `gradle`. Installation of `gradle` is not required. To build, you can run
-```bash
-./gradlew build
-```
+[ˈbʒɛ̃n͇ʧ̑ɨk]
 
-## Testing
-Tests can be run with
-```bash
-./test_all.sh
-```
+A simple programming language with very *polished* syntax.
+
+Implemented by students at Theoretical Computer Science Department of Jagiellonian University
+as the project for *Compilers* course in the 2022/23 winter semester.
+
+The language does not really have anything out of ordinary and it's mostly C-like, with a few somewhat interesting (compared to C) features:
+
+- nested functions able to access outer variables,
+- dynamically allocated arrays with automatic reference counting,
+- generator functions, allowing to pause the execution and pass a value to the caller.
```

Comment:
```suggestion
- nested functions able to access outer variables,
- dynamically allocated arrays orchestrated by automatic reference counting,
- generator functions, capable of pausing their execution and passing a value to the caller.
```

Reply:
"orchestrated by" sounds too poetic to me. The part after "generator functions" is an explanation, not a particularization (it is more or less generally known construct).

Reply:
Decided to changed the wording after "generator functions". Now it is slightly more ambiguous but it sounds more natural.